New Rule  "Do you know how to handle diacritics?" (#8130)
SSWConsulting/SSW.Rules.Content: rules/do-you-know-how-to-handle-diacritics/rule.md

---
type: rule
archivedreason:
title: Do you know how to handle diacritics?
guid: 0d0c5084-5b42-45e0-afb5-c6f886d791ea

> uri: do-you-know-how-to-handle-diacritics

Reviewer: Shorten URI
e.g. handling-diacritics

> created: 2023-09-14T14:38:37.0000000Z

Reviewer: ```suggestion
created: 2024-03-13T14:38:37.0000000Z
```

authors:
  - title: Chloe Lin
    url: https://ssw.com.au/people/chloe-lin
related: []
---

> Imagine you're building a search feature for a multilingual website. A user searches for "résumé" , but the database contains entries without accented characters like "resume." Without normalization, these variations might not match, leading to inaccurate search results.
> 
> Thankfully, JavaScript provides us with a handy method called "normalize()" that helps us tackle this problem effortlessly.
> 
> <!--endintro-->
> 

Reviewer: I think it should start more general , then look at how you can do it in a specific tech (JS then and maybe .NET)

```suggestion
Imagine you're developing a search feature for a multilingual website. When a user searches for a term like "résumé", but the database contains variations like "resume" without accented characters, it leads to missed matches and incomplete search results. Furthermore, this lack of normalization can introduce inconsistencies in sorting and filtering, complicating data analysis and user navigation.

Implementing character normalization by converting accented and special characters to their closest unaccented equivalents allows terms like "résumé" to match "resume" enhancing search accuracy and overall user experience by addressing these multilingual nuances.

This is especially important for internationalization and site serves users worldwide effectively, regardless of linguistic differences.

<!--endintro-->

### Handling diacritics in JS
JavaScript provides us with a handy method called [normalize()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize) that helps us tackle this problem effortlessly.
```

> The method takes an optional parameter specifying the normalization form. The most common form is the Unicode Normalization Form Canonical Deomposition (NFD), which is suitable for most use cases. Let's see how we can use it:
> 

Reviewer: ```suggestion
The method takes an optional parameter specifying the normalization form. The most common form is the Unicode Normalization Form Canonical Deomposition (NFD), which is suitable for most use cases. 
See [Unicode equivalence](https://en.wikipedia.org/wiki/Unicode_equivalence)

Let's see how we can use it:

```

---
type: rule
archivedreason:
title: Do you know how to handle diacritics?
guid: 0d0c5084-5b42-45e0-afb5-c6f886d791ea

> uri: handling-diacritics

Reviewer 2: Please update the folder path to match the uri `handling-diacritics`

Author: Done :)
